### DISCUSSION TITLE
Question: is it possible to `git fetch` with `--force`

### DISCUSSION THREAD

**Discussion by boudewijn-zicht** [magit/magit#4705] [Support] [answered]
Currently there does not seem to be an option to add the `--force` flag when performing a `git fetch`.

We have, for whatever reason, tags with the same name that move periodically around the git tree.  I have to run `git fetch --all --tags --force` to retieve them because the new tags will 'clobber existing tags'.  It would be nice if I could do this from Magit instead of moving to the command line.

From the magit process logs
```
  1 git … fetch --all --tags
Fetching origin
remote: Counting objects: 100% (1/1), done.        
remote: Total 1 (delta 0), reused 1 (delta 0), pack-reused 0        
Unpacking objects: 100% (1/1), 176 bytes | 176.00 KiB/s, done.
From github.com:company/example-repo
 ! [rejected]          foo-tag -> foo-tag  (would clobber existing tag)
error: Could not fetch origin
```

Maybe this exists and I am overlooking it?

**Comment by kyleam** [comment, marked as answer]
`--force` may be worth adding to the fetch transient, likely hidden by default.

Without that, here are the some ways you can handle this while still fetching from Magit:

  * fetch an explicit refspec (`magit-fetch-refspec`), passing `+refs/tags/*:refs/tags/*` as the refspec

    The `+` is equivalent to using `--force`.

  * Configure `+refs/tags/*:refs/tags/*` as an additional refspec for the remote in `$GIT_DIR/config`.

    From a safety perspective, this probably isn't a good idea, but perhaps makes sense depending on the situation.

  * add `--force` yourself in your customization

    ```elisp
    (transient-append-suffix 'magit-fetch "-t"
      '("-f" "Bypass safety checks" "--force"))
    ```

**Reply by boudewijn-zicht** [reply]
Thank you for your quick answer.  I went for the last option to add `--force` to the fetch transient and it works perfectly.  Much appreciated!
